client/src/components/FinanceDashboard/FinanceDashboard.js

```diff
@@ -1,46 +1,30 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import './FinanceDashboard.scss';
 import PropTypes from 'prop-types';
+import { UIcontext } from '../contexts/UIcontext/UIcontext';
 
-class FinanceDashboard extends React.Component {
-    state = {
-        balances: [],
-        errorState: null
-    }
+const FinanceDashboard = (props) => {
+    const { financeDashboard, financePockets } = useContext(UIcontext).dictionary;
+    const { BEFOREPOCKETNAME, AFTERPOCKETNAME } = financeDashboard;
 
-    componentDidMount () {
-        this.buildFinanceOverview(this.props.balance, this.props.currency);
-    }
+    const { balance, currency } = props;
 
-    buildFinanceOverview = (balance, currency) => {
-        try {
-            const categories = [];
-            for (const [key, value] of Object.entries(balance)) {
-                categories.push(
+    return (
+        <div className = "overview" >
+            {Object.entries(balance).map(([key, value]) => {
+                return (
                     <div key = {key}>
-                        {`Current balance of ${key} pocket: ${value} ${currency}`}
+                        {BEFOREPOCKETNAME + financePockets[key.toUpperCase()] + AFTERPOCKETNAME + `${value} ${currency}`}
                     </div>
                 );
-            }
-            this.setState({ balances: categories });
-        } catch (error) {
-            this.props.onError(error);
```

Comment:
Let's ask Benji, what is the purpose of this try/catch & error handling here.
Maybe he know something we can not undestand right now.

Reply:
@JoBenjamin en nem lattam hogy mi celbol van ott a try catch. Miert raktad bele?